File: P389479.py
def solution(players, m, k):
    answer = 0
    servers = [0] * 24
    for i in range(24):
        if players[i] >= (servers[i]+1)*m:
            over = (players[i] - (servers[i]+1)*m) // m
            over += 1
            for j in range(i, i+k):
                if j < 24:
                    servers[j] += over
            answer += over
    return answer

File: P389479.java
public class P389479 {
    public int solution(int[] players, int m, int k) {
        int answer = 0;
        int[] servers = new int[24];
        for (int i=0; i<24; i++) {
            if (players[i] >= (servers[i]+1)*m) {
                int over = (players[i] - (servers[i]+1)*m) / m;
                over += 1;
                for (int j=i; j<i+k; j++) {
                    if (j < 24) servers[j] += over;
                }
                answer += over;
            }
        }
        return answer;
    }
}